### PR TITLE
TMDM-12841 Add unit test.

### PR DESCRIPTION
**Pull request by sbliu**
**What is the current behavior?** (You should also link to an open issue here)
https://jira.talendforge.org/browse/TMDM-12841

**What is the new behavior?**

 Add unit test.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [x] Other... Please describe:
       Add unit test.
**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: test/plugins/org.talend.mdm.repository.test/src/main/java/org/talend/mdm/repository/ui/widgets/UserVarValueValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        List<String> validUserVars = new ArrayList<>();
+
+        validUserVars.add(_PREFIX_USER_VAR + UserField.Id.field + _SUFFIX_USER_VAR);
+        validUserVars.add(_PREFIX_USER_VAR + UserField.First_Name.field + _SUFFIX_USER_VAR);
+        validUserVars.add(_PREFIX_USER_VAR + UserField.Last_Name.field + _SUFFIX_USER_VAR);
+        validUserVars.add(_PREFIX_USER_VAR + UserField.User_Name.field + _SUFFIX_USER_VAR);
+        validUserVars.add(_PREFIX_USER_VAR + UserField.Language.field + _SUFFIX_USER_VAR);
+
+        UserVarValueValidator userVarValueValidator = new UserVarValueValidator();
+        //
```
Comment (VictorLi-talendbj) [inline test/plugins/org.talend.mdm.repository.test/src/main/java/org/talend/mdm/repository/ui/widgets/UserVarValueValidatorTest.java]: We may remove this kind of code. It is not a comment I think.

File: test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        List<WSWhereOperator> operators = new ArrayList<>(Arrays.asList(WSWhereOperator.values()));
+        operators.remove(WSWhereOperator.EMPTY_NULL);
+
+        List<String> conditionValues = new ArrayList<>(operators.size());
+        for (int i = 0; i < operators.size(); i++) {
+            conditionValues.add("");
+        }
+
+        List<WSWhereConditionE> whereCondions = createWhereCondions(operators, conditionValues);
+        //////
```
Comment (VictorLi-talendbj) [inline test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java]: We may remove this kind of code. It is not a comment I think.

File: test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        for (int i = 0; i < operators.size(); i++) {
+            WSWhereConditionE wsWhereConditionE = MdmserverobjectFactory.eINSTANCE.createWSWhereConditionE();
+            WSWhereOperatorE whereOperatorE = MdmserverobjectFactory.eINSTANCE.createWSWhereOperatorE();
+            whereOperatorE.setValue(operators.get(i).name());
+            wsWhereConditionE.setLeftPath("path/" + i);
+            wsWhereConditionE.setOperator(whereOperatorE);
+            wsWhereConditionE.setRightValueOrPath(conditionValues.get(i % size));
+            whereCondions.add(wsWhereConditionE);
+        }
+
```
Comment (VictorLi-talendbj) [inline test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java]: Useless blank line?

File: test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+/**
+ *
+ */
+@SuppressWarnings("nls")
+public class ViewValidatorTest {
+
+    @Test
+    public void testGetViewName() throws Exception {
+        String uri = "file:///D:/workspace/TEST/MDM/view/web/Browse_items_Product_0.1.item";
```
Comment (VictorLi-talendbj) [inline test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java]: Do we also support other kinds of view names? such as: Browse_items_Product$AndFamily_0.1.item

File: test/plugins/org.talend.mdm.repository.test/src/main/java/org/talend/mdm/repository/ui/widgets/UserVarValueValidatorTest.java (hunk 1 above)
Comment (VictorLi-talendbj) [reply test/plugins/org.talend.mdm.repository.test/src/main/java/org/talend/mdm/repository/ui/widgets/UserVarValueValidatorTest.java]: I got it, then I would suggest to add a comment for each one.

File: main/plugins/org.talend.mdm.repository.validate/src/org/talend/mdm/repository/core/validate/view/ViewValidator.java [hunk 5]
```diff
@@ -38,47 +39,66 @@
...
-                        String validateMsg = Messages.bind(Messages.ViewValidator_error, viewName, conditionE.getLeftPath(),
-                                Util.toReadable(WSWhereOperator.fromValue(operator)), userVarValue);
+            validateConditions(viewValidationReport, viewName, whereConditions);
+        }
+        return viewValidationReport;
+    }
+
+    private void validateConditions(ViewValidationReport viewValidationReport, String viewName,
+            List<WSWhereConditionE> whereConditions) {
+        if(whereConditions != null && whereConditions.size() >0) {
```
Comment (lpw82gf) [inline main/plugins/org.talend.mdm.repository.validate/src/org/talend/mdm/repository/core/validate/view/ViewValidator.java]: Need to format

File: test/plugins/org.talend.mdm.repository.test/src/main/java/org/talend/mdm/repository/ui/widgets/UserVarValueValidatorTest.java (hunk 1 above)
Comment (lpw82gf) [reply test/plugins/org.talend.mdm.repository.test/src/main/java/org/talend/mdm/repository/ui/widgets/UserVarValueValidatorTest.java]: Yes, only `//` doesn't make sense, please add real comment for different test blocks.

File: test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+import org.mockito.Mockito;
+import org.mockito.internal.util.reflection.Whitebox;
+import org.talend.mdm.repository.model.mdmserverobject.MdmserverobjectFactory;
+import org.talend.mdm.repository.model.mdmserverobject.WSWhereConditionE;
+import org.talend.mdm.repository.model.mdmserverobject.WSWhereOperatorE;
+import org.talend.mdm.repository.ui.widgets.UserVarValueValidator;
+
+import com.amalto.workbench.webservices.WSWhereOperator;
+
+/**
```
Comment (lpw82gf) [inline test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java]: If no real comment here, the following could be removed.
```
/**
/*
*/
```

File: test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java (hunk 2 above)
Comment (lpw82gf) [reply test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java]: Please add real comment for different test blocks.

File: test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java (hunk 3 above)
Comment (sbliu) [reply test/plugins/org.talend.mdm.repository.validate.test/src/main/java/org/talend/mdm/repository/core/validate/view/ViewValidatorTest.java]: modified

**Pull request merged**